tgg/seqr_scripts/sex_check.py

```diff
@@ -0,0 +1,362 @@
+    sex_mt = sex_mt.annotate_cols(**{f"{chr_name}_mean_dp": hl.agg.mean(sex_mt.DP)})
+    sex_ht = sex_mt.cols()
+    return sex_ht
+
+
+def run_hails_impute_sex(
+    mt: hl.MatrixTable,
+    build: str,
+    out_path: str,
+    xy_fstat_threshold: float = 0.75,
+    xx_fstat_threshold: float = 0.5,
+    aaf_threshold: float = 0.05,
+) -> hl.Table:
+    """
+    Impute sex, annotate MatrixTable with results, and output a histogram of fstat values.
+    :param MatrixTable mt: MatrixTable containing samples to be ascertained for sex
+    :param build: Reference used, either GRCh37 or GRCh38
+    :param out_path: Path to bucket for histogram
+    :param xy_fstat_threshold: F-stat threshold above which a sample will be called XY. Default is 0.75
+    :param xx_fstat_threshold: F-stat threshold below which a sample will be called XX. Default is 0.5
+    :param aaf_threshold: Alternate allele frequency threshold for `hl.impute_sex`. Default is 0.05
+    :return: Table with imputed sex annotations
+    """
+
+    logger.warning(
+        "User needs to decide on fstat thresholds for XY/XX by looking at fstat plots!"
+    )
+
+    # Filter to the X chromosome and impute sex
+    mt = hl.filter_intervals(
+        mt,
+        [
+            hl.parse_locus_interval(
+                get_reference_genome(mt.locus).x_contigs[0], reference_genome=build
+            )
+        ],
+    )
+    sex_ht = hl.impute_sex(
+        mt.GT,
+        aaf_threshold=aaf_threshold,
+        male_threshold=xy_fstat_threshold,
+        female_threshold=xx_fstat_threshold,
+    )
+    mt = mt.annotate_cols(**sex_ht[mt.col_key])
+    sex_ht = mt.cols()
+
+    # Plot histogram of fstat values
+    df = sex_ht.to_pandas()
+    plt.clf()
+    plt.hist(df["f_stat"])
+    plt.xlabel("Fstat")
+    plt.ylabel("Frequency")
+    plt.axvline(xy_fstat_threshold, color="blue", linestyle="dashed", linewidth=1)
+    plt.axvline(xx_fstat_threshold, color="red", linestyle="dashed", linewidth=1)
+
+    with hl.hadoop_open(out_path, "wb") as out:
+        plt.savefig(out)
+
+    return sex_ht
+
+
+def call_sex(
+    callset: str,
+    temp_path: str,
+    out_path: str,
+    use_y_cov: bool = False,
+    add_x_cov: bool = False,
+    y_cov_threshold: float = 0.1,
+    normalization_contig: str = "20",
+    xy_fstat_threshold: float = 0.75,
+    xx_fstat_threshold: float = 0.5,
+    aaf_threshold: float = 0.05,
+    call_rate_threshold: float = 0.25,
+    final_annotations: List[str] = [
+        "is_female",
+        "f_stat",
+        "n_called",
+        "expected_homs",
+        "observed_homs",
+        "sex",
+    ],
+) -> None:
+    """
+    Call sex for the samples in a given callset and export results file to the callset directory.
+    :param str callset: String of full MatrixTable path for the callset
+    :param temp_path: Path to bucket for temporary data
+    :param out_path: Path to bucket for text file of final table
```

Comment:
```suggestion
    :param out_path: Path to bucket for text file of final table and for f-stat histogram
```

Reply:
this value should likely be a bucket name, otherwise the histogram will get written to this path (and be promptly overwritten with the sex text file). for example, you could change `out_path` to `out_bucket` and use it like this for the histogram:
```
out_path = f"{out_bucket}/fstat_histogram.png"  
with hl.hadoop_open(out_path, "wb") as out:
    plt.savefig(out)
```
and for the sex text file:
```
out_path = f"{out_bucket}/{mt_name}_sex.txt" 
sex_ht.export(out_path)
```